### PR TITLE
Resolve with latest TravisCI updates

### PR DESCRIPTION
**Pull request by Lestropie**
Cherry-picking commits from #1642.

Given TravisCI failures will preclude any bug fixes from being merged to `master`, I think the preferable approach in such a circumstance is:

1. Get CI working on `master`;

2. Merge latest updates on `master` into `dev`;

3. Pull requests merging into `dev` remain focused on the issue at hand and don't get conflated with CI experimentation / fixes.

4. Add a "`CI`" label in GitHub so that issues / PRs specific to continuous integration are flagged as such, and e.g. can be ignored when generating changelogs.

### CODE REVIEW

**Comment by Lestropie** [comment]
bba9b51 works on GCC9 on my machine. There was a particular issue [here](https://github.com/MRtrix3/mrtrix3/blob/3.0_RC3/src/gui/opengl/shader.h#L140-L145), which looks suspiciously like a move assignment rather than a copy assignment; it's possible that the implicit copy constructor of that class might lead to unwanted behaviour somewhere like [here](https://github.com/MRtrix3/mrtrix3/blob/3.0_RC3/src/gui/mrview/mode/ortho.h#L37), as multiple instances of the `Program` class could contain the same index value, and any one of them could call `gl::DeleteProgram()` on it, invalidating the underlying data of the other instances.

Basically the changes are to prevent copy-construction or copy-assignment of that `Program` class, and to use `NOMEMALIGN` to make sure that the `std::vector` specialisation isn't invoked for duplicates of the `Projection` class (as definition of a specialisation of `std::vector` requires that the underlying class has a copy constructor, and therefore you can't have the class or any of its derivatives forbidding the copy constructor). None of those classes have underlying Eigen members so I don't think it should cause any issues.

**Comment by Lestropie** [comment]
I can't review since I listed the PR; need somebody else to click the OK button.
